.github/workflows/dart-desktop-tests.yml

```diff
@@ -3,18 +3,10 @@ name: Dart desktop tests
 on:
   workflow_call:
     inputs:
-      os:
```

Comment:
I'm not sure I'm a huge fan of the changes in this file. I understand that it removes some redundant pieces of information, but it complicates the setup somewhat and forces readers to familiarize themselves with the architectures of the various github-hosted runners.

Reply:
The previous code had assumptions about the architecture, that didn't actually match the runners, implicitly assuming arch was `x86_64` unless `architecture` was set to `arm`. That is not correct for macos-14, which is being rolled out as `macos-latest` as of yesterday (https://github.com/github/roadmap/issues/926).

One thing I regret is that I cannot show the arch in the job title, as we don't know what runner is being used yet, so it is only shown in the steps. And obviously I would have preferred for github action expressions to be expressive enough to lowercase a simple string 🙄 

What if I reintroduce the `architecture` argument, and make a step that explicitly test that it actually match `runner.arch`?

Reply:
Sure, I'd be fine with that.